MQClient/queue.py

```diff
@@ -86,57 +107,73 @@ def close(self) -> None:
+        is raised, the message is rejected, but messages continue to
+        be received (this is configured by `self._propagate_recv_error`).
+        Multiple calls to `recv()` and/or recycling an instance are both okay,
+        however if the queue has not been closed (e.g. premature termination
+        of the iterator by a consumer-sider raised exception) the original
+        parameters (`timeout`) are reused.
 
-        Args:
-            timeout (int): seconds to wait idle before stopping (default: 60)
-        Yields:
-            Any: object of data received
+        Example:
+            with queue.recv() as stream:
+                for data in stream:
+                    ...
+
+        Keyword Arguments:
+            timeout {int} -- seconds to wait idle before stopping (default: {60})
+
+        Returns:
+            MessageGeneratorContext -- context manager and generator object
         """
-        for msg in self._backend.message_generator(self.raw_sub_queue, timeout=timeout, propagate_error=False):
-            data = pickle.loads(msg.data)
-            yield data
+        if (not self.message_generator_context) or (not self._sub_queue) or self._sub_queue.was_closed:
+            logging.debug("Creating new MessageGeneratorContext instance.")
+            self.message_generator_context = MessageGeneratorContext(sub=self.raw_sub_queue,
+                                                                     timeout=timeout,
+                                                                     propagate_error=self._propagate_recv_error)
+        return self.message_generator_context
 
     @contextlib.contextmanager
-    def recv_one(self) -> typing.Generator[typing.Any, None, None]:
-        """
-        Receive one message from the queue.
+    def recv_one(self) -> Generator[Any, None, None]:
+        """Receive one message from the queue.
 
-        This is a context manager. If an exception is raised, the message
-        is rejected.
+        This is a context manager. If an exception is raised, the message is rejected.
 
-        Returns:
-            Any: object of data received, or None if queue is empty
+        Decorators:
+            contextlib.contextmanager
+
+        Yields:
+            Any -- object of data received, or None if queue is empty
+
+        Raises:
+            Exception -- [description]
         """
-        msg = self._backend.get_message(self.raw_sub_queue)
+        msg = self.raw_sub_queue.get_message()
         if not msg:
             raise Exception('No message available')
         try:
             yield pickle.loads(msg.data)
         except Exception:
-            self._backend.reject_message(self.raw_sub_queue, msg.msg_id)
+            self.raw_sub_queue.reject_message(msg.msg_id)
             raise
         else:
-            self._backend.ack_message(self.raw_sub_queue, msg.msg_id)
+            self.raw_sub_queue.ack_message(msg.msg_id)
```

Comment:
Queue.recv_one is always `auto_ack` then?

Reply:
Yeah. Well everything from Queue's POV is `auto_ack`